tutorials/hyperspectral/RasterStack-RGB-Images-in-R-Using-HSI.Rmd

```diff
@@ -236,61 +211,61 @@ band2Raster <- function(file, band, noDataValue, xMin, yMin, res, crs){
 Now that the function is created, we can create our list of rasters. The list 
 specifies which bands (or dimensions in our hyperspectral dataset) we want to 
 include in our raster stack. Let's start with a typical RGB (red, green, blue) 
-combination. We will use bands 58, 34, and 19. 
+combination. We will use bands 14, 9, and 4 (bands 58, 34, and 19 in a full 
+NEON hyperspectral dataset).
 
 <div id="ds-dataTip" markdown="1">
 <i class="fa fa-star"></i>**Data Tip - wavelengths and bands:** Remember that 
-you can look at the wavelengths dataset to determine the center wavelength value 
-for each band. 
+you can look at the wavelengths dataset in the HDF5 file to determine the center 
+wavelength value for each band. 
 </div>
 
 
 ```{r create-raster-stack }
-
 #create a list of the bands we want in our stack
-rgb <- list(58,34,19)
+rgb <- list(14,9,4) #list(58,34,19) when using full NEON hyperspectral dataset
 #lapply tells R to apply the function to each element in the list
-rgb_rast <- lapply(rgb,band2Raster, file = f, 
+rgb_rast <- lapply(rgb,FUN=band2Raster, file = f,
                    noDataValue=myNoDataValue, 
-                   xMin=xMin, yMin=yMin, res=1,
-                   crs=myCrs)
+                   extent=rasExt,
+                   CRS=myCRS)
 
 #check out the properties or rgb_rast
 #note that it displays properties of 3 rasters.
 
 rgb_rast
 
 #finally, create a raster stack from our list of rasters
-hsiStack <- stack(rgb_rast)
+rgbStack <- stack(rgb_rast)
 
 ```
 
-<a href="http://www.r-bloggers.com/using-apply-sapply-lapply-in-r/" target="_blank">More about Lapply here</a>. 
+In the code chunk above, we used the `lapply()` function, which is a powerful,
+flexible way to apply a function (in this case, our `band2Raster()` fucntion)
+multiple times. You can <a href="http://www.r-bloggers.com/using-apply-sapply-lapply-in-r/" target="_blank">learn more about lapply() here</a>. 
 
 NOTE: We are using the `raster stack` object in R to store several rasters that 
-are of the same CRS and extent.
-
+are of the same CRS and extent. This is a popular and convenient way to organize 
+co-incident rasters.
 
 Next, add the names of the bands to the raster so we can easily keep track of 
 the bands in the list.
 
 ```{r plot-raster-stack }
-
-#Add the band numbers as names to each raster in the raster list
-
 #Create a list of band names
 bandNames <- paste("Band_",unlist(rgb),sep="")
 
-names(hsiStack) <- bandNames
+# set the rasterStack's names equal to the list of bandNames created above
+names(rgbStack) <- bandNames
+
 #check properties of the raster list - note the band names
-hsiStack
+rgbStack
 
 #scale the data as specified in the reflInfo$Scale Factor
-
-hsiStack <- hsiStack/reflInfo$`Scale Factor`
+rgbStack <- rgbStack/as.integer(reflInfo$Scale_Factor)
 
 ### Plot one raster in the stack to make sure things look OK.
```

Comment:
Not sure why this has three ###, simply having # makes more sense.

Reply:
Good call - reduced to one #